core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -565,12 +565,17 @@ import scala.util.control.NonFatal
 
   private def commitAggregatedOffsets(): Unit = if (commitMaps.nonEmpty && !rebalanceInProgress) {
     val aggregatedOffsets = aggregateOffsets(commitMaps)
-    commitRefreshing.add(aggregatedOffsets)
+    // commits can occur after the partition has been unassigned from the consumer, so ensure that we only attempt to
+    // commit partitions that are currently assigned to the consumer. For high volume topics, this can lead to small
+    // amounts of replayed data during a rebalance, but for low volume topics can ensure that consumers never appear
+    // 'stuck' because of out-of-order commits from slow consumers.
+    val assignedOffsetToCommit = aggregatedOffsets.filterKeys(consumer.assignment().contains).toMap
+    commitRefreshing.add(assignedOffsetToCommit)
     val replyTo = commitSenders
     // flush the data before calling `consumer.commitAsync` which might call the callback synchronously
     commitMaps = List.empty
     commitSenders = Vector.empty
-    commit(aggregatedOffsets, replyTo)
+    commit(assignedOffsetToCommit, replyTo)
```

Comment:
Yeah, you are right. I was thinking that b/c we filter out revoked before committed, it should be fine, but it could be that the partition is revoked between making the commit and updating the deadline.

Unfortunately, it is going to be very hard to test that specific behavior. But i'll go back to the original fix here to so that we keep only refreshing commits that we own.